go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -81,6 +81,10 @@ func (vr *vreplicator) Replicate(ctx context.Context) error {
 	vr.tableKeys = tableKeys
 
 	for {
+		// This rollback is a no-op. It's here for safety
+		// in case the functions below leave transactions open.
+		vr.dbClient.Rollback()
```

Comment:
It seems that we expect that at this point, we are not in a transaction. Do you think it is helpful to log a warning if this expectation is violated?

Reply:
It's not worth it. The high level model is that this call is sufficient, and the lower level functions don't need to rollback. But right now, they rollback only as extra precaution.